### ISSUE TITLE
Loss turns nan with plaidml backend

### ISSUE THREAD

**Issue by ZagButNoZig** [plaidml/plaidml#1029]
PlaidML version = 0.7.0
Keras version = 2.2.4
Device = AMD Radeon HD 6800 

When training my model the loss turn nan after the first step. On CPU backend everything works fine.

Here is my model architecture: 

`__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            (None, 15, 8, 8)     0                                            
__________________________________________________________________________________________________
input_conv (Conv2D)             (None, 10, 8, 8)     1350        input_1[0][0]                    
__________________________________________________________________________________________________
input_batchnorm (BatchNormaliza (None, 10, 8, 8)     40          input_conv[0][0]                 
__________________________________________________________________________________________________
input_relu (Activation)         (None, 10, 8, 8)     0           input_batchnorm[0][0]            
__________________________________________________________________________________________________
res_conv1-0 (Conv2D)            (None, 10, 8, 8)     900         input_relu[0][0]                 
__________________________________________________________________________________________________
res_batchnorm10 (BatchNormaliza (None, 10, 8, 8)     40          res_conv1-0[0][0]                
__________________________________________________________________________________________________
res_relu10 (Activation)         (None, 10, 8, 8)     0           res_batchnorm10[0][0]            
__________________________________________________________________________________________________
res_conv20 (Conv2D)             (None, 10, 8, 8)     900         res_relu10[0][0]                 
__________________________________________________________________________________________________
res_batchnorm20 (BatchNormaliza (None, 10, 8, 8)     40          res_conv20[0][0]                 
__________________________________________________________________________________________________
res_add0 (Add)                  (None, 10, 8, 8)     0           input_relu[0][0]                 
                                                                 res_batchnorm20[0][0]            
__________________________________________________________________________________________________
res_relu20 (Activation)         (None, 10, 8, 8)     0           res_add0[0][0]                   
__________________________________________________________________________________________________
res_conv1-1 (Conv2D)            (None, 10, 8, 8)     900         res_relu20[0][0]                 
__________________________________________________________________________________________________
res_batchnorm11 (BatchNormaliza (None, 10, 8, 8)     40          res_conv1-1[0][0]                
__________________________________________________________________________________________________
res_relu11 (Activation)         (None, 10, 8, 8)     0           res_batchnorm11[0][0]            
__________________________________________________________________________________________________
res_conv21 (Conv2D)             (None, 10, 8, 8)     900         res_relu11[0][0]                 
__________________________________________________________________________________________________
res_batchnorm21 (BatchNormaliza (None, 10, 8, 8)     40          res_conv21[0][0]                 
__________________________________________________________________________________________________
res_add1 (Add)                  (None, 10, 8, 8)     0           res_relu20[0][0]                 
                                                                 res_batchnorm21[0][0]            
__________________________________________________________________________________________________
res_relu21 (Activation)         (None, 10, 8, 8)     0           res_add1[0][0]                   
__________________________________________________________________________________________________
res_conv1-2 (Conv2D)            (None, 10, 8, 8)     900         res_relu21[0][0]                 
__________________________________________________________________________________________________
res_batchnorm12 (BatchNormaliza (None, 10, 8, 8)     40          res_conv1-2[0][0]                
__________________________________________________________________________________________________
res_relu12 (Activation)         (None, 10, 8, 8)     0           res_batchnorm12[0][0]            
__________________________________________________________________________________________________
res_conv22 (Conv2D)             (None, 10, 8, 8)     900         res_relu12[0][0]                 
__________________________________________________________________________________________________
res_batchnorm22 (BatchNormaliza (None, 10, 8, 8)     40          res_conv22[0][0]                 
__________________________________________________________________________________________________
res_add2 (Add)                  (None, 10, 8, 8)     0           res_relu21[0][0]                 
                                                                 res_batchnorm22[0][0]            
__________________________________________________________________________________________________
res_relu22 (Activation)         (None, 10, 8, 8)     0           res_add2[0][0]                   
__________________________________________________________________________________________________
res_conv1-3 (Conv2D)            (None, 10, 8, 8)     900         res_relu22[0][0]                 
__________________________________________________________________________________________________
res_batchnorm13 (BatchNormaliza (None, 10, 8, 8)     40          res_conv1-3[0][0]                
__________________________________________________________________________________________________
res_relu13 (Activation)         (None, 10, 8, 8)     0           res_batchnorm13[0][0]            
__________________________________________________________________________________________________
res_conv23 (Conv2D)             (None, 10, 8, 8)     900         res_relu13[0][0]                 
__________________________________________________________________________________________________
res_batchnorm23 (BatchNormaliza (None, 10, 8, 8)     40          res_conv23[0][0]                 
__________________________________________________________________________________________________
res_add3 (Add)                  (None, 10, 8, 8)     0           res_relu22[0][0]                 
                                                                 res_batchnorm23[0][0]            
__________________________________________________________________________________________________
res_relu23 (Activation)         (None, 10, 8, 8)     0           res_add3[0][0]                   
__________________________________________________________________________________________________
res_conv1-4 (Conv2D)            (None, 10, 8, 8)     900         res_relu23[0][0]                 
__________________________________________________________________________________________________
res_batchnorm14 (BatchNormaliza (None, 10, 8, 8)     40          res_conv1-4[0][0]                
__________________________________________________________________________________________________
res_relu14 (Activation)         (None, 10, 8, 8)     0           res_batchnorm14[0][0]            
__________________________________________________________________________________________________
res_conv24 (Conv2D)             (None, 10, 8, 8)     900         res_relu14[0][0]                 
__________________________________________________________________________________________________
res_batchnorm24 (BatchNormaliza (None, 10, 8, 8)     40          res_conv24[0][0]                 
__________________________________________________________________________________________________
res_add4 (Add)                  (None, 10, 8, 8)     0           res_relu23[0][0]                 
                                                                 res_batchnorm24[0][0]            
__________________________________________________________________________________________________
res_relu24 (Activation)         (None, 10, 8, 8)     0           res_add4[0][0]                   
__________________________________________________________________________________________________
value_conv-1-4 (Conv2D)         (None, 4, 8, 8)      40          res_relu24[0][0]                 
__________________________________________________________________________________________________
value_batchnorm (BatchNormaliza (None, 4, 8, 8)      16          value_conv-1-4[0][0]             
__________________________________________________________________________________________________
value_relu (Activation)         (None, 4, 8, 8)      0           value_batchnorm[0][0]            
__________________________________________________________________________________________________
value_flatten (Flatten)         (None, 256)          0           value_relu[0][0]                 
__________________________________________________________________________________________________
value_dense (Dense)             (None, 48)           12336       value_flatten[0][0]              
__________________________________________________________________________________________________
value_out (Dense)               (None, 1)            49          value_dense[0][0]                
==================================================================================================
Total params: 23,231
Trainable params: 23,003
Non-trainable params: 228
__________________________________________________________________________________________________`

Its a resnet. And interestingly this issue only happens when the amount of res layers > 5.
